learning/katas/go/Core Transforms/Combine/CombineFn/pkg/task/task.go

```diff
@@ -0,0 +1,55 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package task
+
+import (
+	"github.com/apache/beam/sdks/go/pkg/beam"
+)
+
+func ApplyTransform(s beam.Scope, input beam.PCollection) beam.PCollection {
+	return beam.Combine(s, &averageFn{}, input)
+}
+
+type averageAccum struct {
+	Count int64
+	Sum float64
+}
+
+type averageFn struct{}
+
+func (c *averageFn) CreateAccumulator() averageAccum {
+	return averageAccum{}
+}
+
+func (c *averageFn) AddInput(accum averageAccum, input int) averageAccum {
+	accum.Count++
+	accum.Sum += float64(input)
```

Comment:
Why should the input be casted to float?
I think since the inputs are all int, the Sum can be int?